### DISCUSSION TITLE
How to run job at 23:59:59 on the last day of each month

### DISCUSSION THREAD

**Discussion by binbjz** [agronholm/apscheduler#650] [Q&A]
Hi @agronholm ,
    I can run job on the last day of each month. but how can I run job at 23:59:59 on the last day of each month. Thanks.
```python
    job_stores = {
        "default": MemoryJobStore()
    }

    executors = {
        "processpool": ProcessPoolExecutor(max_workers=6)
    }

    def bg_run():
    """
    Scheduler to run in the background inside the application
    """
    scheduler = BackgroundScheduler(jobstores=job_stores, executors=executors)
    scheduler.add_job(main, "cron", year="*", month="*", day="last", id="gl_job")
    scheduler.print_jobs()
    scheduler.start()

    try:
        while True:
            pass
    except (KeyboardInterrupt, SystemExit):
        scheduler.shutdown(wait=False)
```

**Comment by binbjz** [comment]
Use the following expression to check if it works.
```python
    scheduler.add_job(main, "cron", year="*", month="*", day="last",
                      hour="23", minute="59", second="59", id="gl_job")
```

**Reply by agronholm** [reply]
Sounds like you answered your own question? By the way, `year="*", month="*"` can be omitted.

**Reply by agronholm** [reply]
Also, do use `BlockingScheduler` instead of having a busy-wait loop with `while True: pass`.

**Reply by binbjz** [reply]
Thanks for your help.

**Reply by binbjz** [reply]
Thanks for your help. I have a question about BackgroundScheduler.
If I use the BackgroundScheduler and do not add while True: pass, the program will exit after it is executed, and I also find that the program is not executed in the background. I don't understand why. For example, when I execute "rm -rf /tmp/debug.log", the debug.log file is not deleted after the program is executed with BackgroundScheduler. But if I add while True: pass to keep the program going, debug.log will be deleted.

I want to know, how do I correctly make the BackgroundScheduler run a program.

**Reply by agronholm** [reply]
Like I already said, use `BlockingScheduler` instead of `BackgroundScheduler`. This is what it's for.
